packages/gatsby/src/query/graphql-runner.ts

```diff
@@ -177,7 +177,7 @@ export class GraphQLRunner {
       const result =
         errors.length > 0
           ? { errors }
-          : execute({
+          : await execute({
```

Comment:
This will change the dynamics of query execution. Before this change, the `scheduleClearCache` function would've been called immediately after the `execute` call. Now it waits for `execute` to finish.

I have no idea how it will actually impact query running in real projects. Just want to bring your attention as this is not just a syntax change and definitely not a `chore`.

Reply:
This was necessary to make `GATSBY_EXPERIMENTAL_QUERY_CONCURRENCY=1` work at all

Reply:
Hm, it should be working unless I am missing something. How can I repro the problem with `GATSBY_EXPERIMENTAL_QUERY_CONCURRENCY=1`?

Reply:
With the current code that's harder since there are more cases that need to be fixed (working on that, this is one of them). 

Easiest way to check is to run in =1 mode and then do a console.time before/after a certain async op that will be ran multiple times (like page creation, query running, etc). If you get terminal errors that state that timer already exists, you know you're not running in serial.

Alternatively, you can set a global var that counts up before and down afterwards and prints the current value. If it goes above 1, you know you're not serial.

I'll double check tomorrow whether this `await` is indeed necessary. There's a bunch of cases in the chain missing that make serial mode impossible.

Reply:
I realized that you may be right, actually, as long as the promise is returned. I'll update this major PR :p